llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -16210,13 +16248,20 @@ SDValue SITargetLowering::performSubCombine(SDNode *N,
       return Folded;
   }
 
-  if (VT != MVT::i32)
-    return SDValue();
-
   SDLoc SL(N);
   SDValue LHS = N->getOperand(0);
   SDValue RHS = N->getOperand(1);
 
+  if (VT == MVT::i8)
+    if (isNullConstant(LHS) && RHS->getOpcode() == ISD::ABS)
+      return DAG.getNode(ISD::TRUNCATE, SL, MVT::i8,
+                         DAG.getNode(ISD::SUB, SL, MVT::i32,
+                                     DAG.getConstant(0, SL, MVT::i32),
+                                     lowerABSi16(RHS, DAG).getOperand(0)));
```

Comment:
There's a DAG.getNegative helper for this pattern

Reply:
Thanks, I'm using `getNegative` in the real PR.